packages/job-worker/src/peripheralDevice.ts

```diff
@@ -8,8 +8,12 @@ export async function executePeripheralDeviceFunction(
 	context: JobContext,
 	deviceId: PeripheralDeviceId,
 	timeoutTime0: number | null,
-	functionName: string,
-	...args: any[]
+	action: {
```

Comment:
I think this signature makes more sense than the spread operator it used to be.

Perhaps this could be written as `action: Pick<PeripheralDeviceCommand, 'functionName' | 'args' | 'actionId' | 'payload'>`.
That would help ensure the types stay more in sync.

But also, `functionName` and `args` feel like they should be required properties here.
Combining that into the above can be done, but starts to feel messy action: SetRequired<Pick<PeripheralDeviceCommand, 'functionName' | 'args' | 'actionId' | 'payload'>, 'functionName | 'args'>`.

Reply:
if we want to have required properties I think we should separate these into separate methods, because the consumer will use _either_ functionName _or_ actionId.